src/protocols/secure_channel/CASESession.cpp

```diff
@@ -207,32 +210,53 @@ class CASESession::WorkHelper
     // Cancel the work, by clearing the associated session.
     void CancelWork() { mSession.store(nullptr); }
 
+    bool IsCancelled() const { return mSession.load() == nullptr; }
+
 private:
+    // Create a work helper using the specified session, work callback, after work callback, and data (template arg).
+    // Lifetime is not managed, see `Create` for that option.
+    WorkHelper(CASESession & session, WorkCallback workCallback, AfterWorkCallback afterWorkCallback) :
+        mSession(&session), mWorkCallback(workCallback), mAfterWorkCallback(afterWorkCallback)
+    {}
+
     // Handler for the work callback.
     static void WorkHandler(intptr_t arg)
     {
         auto * helper = reinterpret_cast<WorkHelper *>(arg);
-        bool cancel   = false;
-        VerifyOrExit(helper->mSession.load(), ;); // cancelled by `CancelWork`?
-        helper->mStatus = helper->mWorkCallback(helper->mData, cancel);
-        VerifyOrExit(!cancel, ;);                 // canceled by `mWorkCallback`?
-        VerifyOrExit(helper->mSession.load(), ;); // cancelled by `CancelWork`?
-        SuccessOrExit(DeviceLayer::PlatformMgr().ScheduleWork(AfterWorkHandler, reinterpret_cast<intptr_t>(helper)));
-        return;
-    exit:
-        helper->mStrongPtr.reset();
+        // Hold strong ptr while work is handled
+        auto strongPtr(std::move(helper->mStrongPtr));
+        if (!helper->IsCancelled())
```

Comment:
This is fine logic-wise, but why the nested ifs instead of just early returns?

Reply:
Removed all the nesting.